File: 0274-h-index/0274-h-index.java
class Solution {
    public int hIndex(int[] citations) {

        Arrays.sort(citations);

        int result = 0;
        for(int i = 0; i < citations.length; i++){
            if(i + 1 <= citations[citations.length - 1 - i]) {
                result = i + 1;
            }
            else return result;
        }
        
        return result;
    }
}

File: 0274-h-index/0274-h-index.go
func hIndex(citations []int) int {
    all := make([]int, 10000)
    maxValue := 0
    
    for i:= 0; i < len(citations); i++ {
        for citations[i] - 1 >= 0 {
            citations[i]--
            all[citations[i]]++
        }
    }
    
    for i:= 0; i < len(all); i++ {
        if all[i] >= i + 1 {
            maxValue = i + 1
        }
    }
    
    
    return maxValue
}